src/test/lib/storage/buffer/frame_test.cpp

```diff
@@ -0,0 +1,150 @@
+#include "base_test.hpp"
+#include "storage/buffer/frame.hpp"
+
+namespace hyrise {
+
+class FrameTest : public BaseTest {};
```

Comment:
There are plenty of (Debug)Asserts in the Frame code. I think we should test them as well.

Reply:
Pick here, `TEST_F(XYTest, TestAbc)` is a bit too much for me. At least `TestAbc` could be just `Abc` I guess.

Reply:
Furthermore, a lot of the Frame code focusses on parallel execution. I think some stress tests for these parts (`stress_test.cpp` are a good idea).

Reply:
What's the point of testing DebugAsserts? they are basically tests themselves (or invariants).

Reply:
I have a stresstest ready for the whole buffer manager: https://github.com/nikriek/hyrise-fork/blob/3f847898c97925ca21e56e7c3594c8cd77b59a6a/src/test/lib/concurrency/buffer_manager_stress_test.cpp#L17

Reply:
Proper stress tests also came to my mind when seeing the code.
I still have to dig into the BufferManager, but I would assume that buffering could hide some cases where might have concurrency issues because there are other overheads.

What I want to ask is: sure we don't want a stress test solely for modifying Frames?

Reply:
Ok noted.